test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -53,6 +62,50 @@
 public abstract class MapperTestCase extends MapperServiceTestCase {
     protected abstract void minimalMapping(XContentBuilder b) throws IOException;
 
+    protected void field(XContentBuilder builder) throws IOException {
+        builder.field("field");
```

Comment:
I think the only mappers that need to override this are ones that do not extend ConcreteFieldType, so maybe we can add a check in the test and take a different branch if we know we need to do something different here?

Reply:
I am overriding this at the moment  in constant keyword and runtime field mapper. there are concrete field types (for instance the ones that throw when calling existsQuery) that don't subclass ConcreteFieldType, which is partly why I don't like to have that base class. 

This method is really for the field types that should not write the field in the docs at all. Can you expand a bit on what you are suggesting, I am not sure I am getting it?

Reply:
> there are concrete field types (for instance the ones that throw when calling existsQuery) that don't subclass ConcreteFieldType

ah ok, never mind then, I think I misunderstood what was going on